File: src/main/java/com/example/kyudo_app/config/CorsConfig.java
package com.example.kyudo_app.config;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.cors.CorsConfiguration;
import org.springframework.web.cors.UrlBasedCorsConfigurationSource;
import org.springframework.web.filter.CorsFilter;

import java.util.Arrays;

@Configuration
public class CorsConfig {

    @Bean
    public CorsFilter corsFilter() {
        UrlBasedCorsConfigurationSource source = new UrlBasedCorsConfigurationSource();
        CorsConfiguration config = new CorsConfiguration();

        config.setAllowedOrigins(Arrays.asList("http://localhost:5173")); // フロントエンドのオリジンを許可
        config.setAllowedMethods(Arrays.asList("GET", "POST", "PUT", "DELETE", "OPTIONS")); // 許可するメソッド
        config.setAllowedHeaders(Arrays.asList("*")); // すべてのヘッダーを許可
        config.setAllowCredentials(true); // クッキーや認証情報を許可

        source.registerCorsConfiguration("/**", config);
        return new CorsFilter(source);
    }
}

File: src/main/kotlin/com/example/kyudo_app/config/CorsConfig.kt
package com.example.kyudo_app.config

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.web.cors.CorsConfiguration
import org.springframework.web.cors.UrlBasedCorsConfigurationSource
import org.springframework.web.filter.CorsFilter

@Configuration
open class CorsConfig {
    @Bean
    open fun corsFilter(): CorsFilter {
        val source = UrlBasedCorsConfigurationSource()
        val config = CorsConfiguration()

        config.allowedOrigins = listOf(
            "http://localhost:5173",       // IPv4
            "http://[::1]:5173",           // IPv6
            "http://13.113.236.120",       // フロントエンドEC2
            "https://kyudo-record.net"     // 本番ドメイン
        ) // フロントのオリジン
        config.allowedMethods = listOf("GET", "POST", "PUT", "DELETE", "OPTIONS")
        config.allowedHeaders = listOf("*")
        config.allowCredentials = true

        source.registerCorsConfiguration("/**", config)
        return CorsFilter(source)
    }
}
